Makefile

```
@@ -131,12 +131,15 @@ perlcritic: tools/lib/
 .PHONY: test-unused-modules-changed
 test-unused-modules-changed:
 	@echo "[make] Unused modules check called over modified/new files only. For a full run use make test-unused-modules-full"
-	tools/detect_unused_modules -m `git --no-pager diff --name-only --diff-filter=d origin/master | grep '^tests/*' | grep -v '^tests/test_pods/'`
-	tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master products/* | grep -oP "^-.*loadtest.*[\"']\K[^\"'].+(?=[\"'])"`
-	tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master schedule/* | grep -oP "^-\s+- [\"']?\K.*(?=[\"']?)" | grep -v '{{'`
-
-.PHONY: test-unused-modules
-test-unused-modules:
+	export f=$$(mktemp); \
+	git --no-pager diff --name-only --diff-filter=d origin/master | grep '^tests/*' | grep -v '^tests/test_pods/' > $$f; \
+	git --no-pager diff --unified=0 origin/master products/* | sed -n "s~^-.*loadtest\s\+\([\"']\)\([^\"']\+\)\1.*~tests/\2.pm~p" >> $$f; \
+	git --no-pager diff --unified=0 origin/master schedule/* | sed -n "s~^-\s\+-\s\+\([\"']\)\([^\"']\+\)\1.*~tests/\2.pm~p" | grep -v '{{' >> $$f; \
+	sort -u $$f -o $$f; \
+	tools/detect_unused_modules -m $$(cat $$f)
```

How about

```suggestion
	tools/detect_unused_modules -m $((git --no-pager diff --name-only --diff-filter=d origin/master | grep '^tests/*' | grep -v '^tests/test_pods/' && \
	git --no-pager diff --unified=0 origin/master products/* | sed -n "s~^-.*loadtest\s\+\([\"']\)\([^\"']\+\)\1.*~tests/\2.pm~p" && \
	git --no-pager diff --unified=0 origin/master schedule/* | sed -n "s~^-\s\+-\s\+\([\"']\)\([^\"']\+\)\1.*~tests/\2.pm~p" | grep -v '{{') \
	| sort -u)
```

not needing a temporary file (possibly tiny bit faster even)

---

My main concern is that pipes are hard to debug (you error in one big line) and frankly speaking openqa CI shell scripts are really very hard to read already. That's why I slightly prefer using the file.

Also the syntax is wrong: `$((` must be ended by `))` (understand it was just an example and I also had a problem to use `$(( .. ))` I needed to replace it with ` `` `).

`&&` mean if first `git grep` finds nothing, it exits 1 and thus other searches aren't run, thus we'd have to use `;` (I wanted to have single run, because I don't want to have files detected 2x or even 3x :).)

BTW I worried if it's wise to post input to the script, due max line limits.
I tested it on dd379ea6bbe591f143f971f9e4f6005dfe996c9e (MR#13246) which has 1644 lines. Putting that into single line it's 61904 chars. But max line is still higher:
```
$ getconf ARG_MAX 
2097152
```

---

So then the use of pipe is no problem :) you can also use the read command if you prefer. I just think handling a file is overkill. Consider my suggestion to move to the shell script instead

---

I removed `mktemp` yesterday. If you mean to move these `git --no-pager diff` to script, I could do that (having .e.g `-c` for finding modules with these 3 git diff and keep `-m` to provide extra module). But code would be just hidden in the script - IMHO not much difference. And ubuntu on GitHub Actions has even higher limit (4194304), we don't have to worry about the limit for passing to the script.